nwastdlib/asyncio_cache.py

```diff
@@ -0,0 +1,134 @@
+#  Copyright 2019 SURF.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#          http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+import hashlib
+import hmac
+import pickle  # noqa S403
+import sys
+from functools import wraps
+from typing import Any, Callable
+
+import structlog
+from redis.asyncio import Redis as AIORedis
+
+logger = structlog.get_logger(__name__)
+
+
+def deserialize(data: Any) -> Any:
+    return pickle.loads(data)  # noqa S403
+
+
+def serialize(data: Any) -> Any:
+    return pickle.dumps(data)  # noqa S403
+
```

Comment:
Agree on Any for the original data, but the serialized data should be bytes? 

```suggestion
def deserialize(data: bytes) -> Any:
    return pickle.loads(data)  # noqa S403


def serialize(data: Any) -> bytes:
    return pickle.dumps(data)  # noqa S403

```

Reply:
Moved to Interface methods. Will re-assess before merge.

Reply:
Still not sure; I think to types that should work: `Byte`, `ByteArray` and `String`
But it also depends on the used Protocol. Maybe be better to be more explicit in the places where you define an implementation. Like it did in the last commit in this PR.